File: src/Ebay/Server.Application.Legacy/Abstractions/Repositories/IMatchedPairDifferenceRepository.cs
using Server.Domain.Measurements;

namespace Server.Application.Abstractions.Repositories;

public interface IMatchedPairDifferenceRepository : IRepository<MatchedPairDifference, MatchedPairDifferenceId>
{
    Task RemoveByMeasurementIds(IReadOnlySet<string> measurementIds, CancellationToken cancellationToken);
}

File: src/Ebay/Server.Application.New/Abstractions/Repositories/IMatchedPairDifferenceRepository.fs
namespace Server.Application.New.Abstractions.Repositories

open System.Collections.Generic
open System.Threading
open System.Threading.Tasks
open Server.Application.New.Abstractions
open Server.Domain.Measurements

/// <summary>
/// Репозиторий агрегата разницы между замерами для задач подбора пар.
/// </summary>
type IMatchedPairDifferenceRepository =
    inherit IRepository<MatchedPairDifference, MatchedPairDifferenceId>

    /// <summary>
    /// Удаляет все записи разниц, связанные с указанными идентификаторами замеров.
    /// </summary>
    /// <param name="measurementIds">Идентификаторы замеров.</param>
    /// <param name="cancellationToken">Токен отмены операции.</param>
    abstract RemoveByMeasurementIds:
        measurementIds: IReadOnlySet<string> *
        cancellationToken: CancellationToken
            -> Task
